R/wrap_vfold.R

```diff
@@ -0,0 +1,218 @@
+#' A resample of the analysis data consists of V-1 of the folds while the
+#' assessment set contains the final fold.
+#' This function adds the ability to apply an inclusion radius and exclusion
+#' buffer to the assessment set, ensuring that your analysis data is spatially
+#' separated from the assessment set.
+#' In basic V-fold cross-validation (i.e. no repeats), the number of resamples
+#' is equal to V.
+#'
+#' @details
+#' When `radius` and `buffer` are both `NULL`, `spatial_buffer_vfold_cv`
+#' is equivalent to [rsample::vfold_cv()] and `leave_location_out_cv` is
+#' equivalent to [rsample::group_vfold_cv()].
+#'
+#' @inheritParams rsample::vfold_cv
+#' @inheritParams rsample::group_vfold_cv
+#' @param group A variable in data (single character or name) used to create
+#' folds. For leave location out CV, this should be a variable containing
+#' the locations to group observations by, for leave time out CV the
+#' time blocks to group by, and for leave location and time out the
+#' spatiotemporal blocks to group by.
+#' @inheritParams buffer_indices
+#'
+#' @references
+#'
+#' K. Le Rest, D. Pinaud, P. Monestiez, J. Chadoeuf, and C. Bretagnolle. 2014.
+#' "Spatial leave-one-out cross-validation for variable selection in the
+#' presence of spatial autocorrelation," Global Ecology and Biogeography 23,
+#' pp. 811-820, doi: 10.1111/geb.12161.
+#'
+#' H. Meyer, C. Reudenbach, T. Hengl, M. Katurji, and T. Nauss. 2018.
+#' "Improving performance of spatio-temporal machine learning models using
+#' forward feature selection and target-oriented validation,"
+#' Environmental Modelling & Software 101, pp. 1-9,
+#' doi: 10.1016/j.envsoft.2017.12.001.
+#'
+#' @rdname spatial_vfold
+#'
+#' @examplesIf sf::sf_use_s2() && rlang::is_installed("modeldata")
+#' boston_vfold <- spatial_buffer_vfold_cv(
+#'   boston_canopy,
+#'   buffer = 500,
+#'   radius = NULL
+#' )
+#' data(ames, package = "modeldata")
+#' ames_sf <- sf::st_as_sf(ames, coords = c("Longitude", "Latitude"), crs = 4326)
+#' ames_neighborhoods <- spatial_leave_location_out_cv(ames_sf, Neighborhood)
+#'
+#' @export
+spatial_buffer_vfold_cv <- function(data,
+                                    v = 10,
+                                    radius,
+                                    buffer,
+                                    repeats = 1,
+                                    strata = NULL,
+                                    breaks = 4,
+                                    pool = 0.1,
+                                    ...) {
+
+  if (missing(radius) || missing(buffer)) {
+    use_vfold <- NULL
+    if (missing(radius) && missing(buffer)) {
+      use_vfold <- c(i = "Or use `rsample::vfold_cv() to use a non-spatial V-fold")
+    }
+    rlang::abort(
+      c(
+        "`spatial_buffer_vfold_cv()` requires both `radius` and `buffer` be provided",
+        i = "Set either `radius` or `buffer` to NULL to not use that argument",
```

Comment:
What would be a better way to word "not use that argument"? I mean, to clarify what they'll get with `NULL` for those? I feel confused about it when reading the message as is. 😔

Reply:
Something like:
> Use `NULL` for resampling without one of `radius` or `buffer`, like `radius = NULL, buffer = 5000`